src/merkle/store.rs

```diff
@@ -280,6 +278,7 @@ impl MerkleStore {
             }
         }
 
+        path.reverse();
         Ok((node.into(), MerklePath::new(path)))
```

Comment:
We return `node.into()` here, but it is always set to `RpoDigest::default()`. What is the original intent? To be the leaf or the target root?

Reply:
the path doesn't include the root, that should have been it, the shadowing of the `node` variable was a bug.